third_party/maya/lib/usdMaya/usdWriteJobCtx.h

```diff
@@ -0,0 +1,58 @@
+#ifndef PXRUSDMAYA_USDWRITEJOBCTX_H
+#define PXRUSDMAYA_USDWRITEJOBCTX_H
+
+#include "pxr/pxr.h"
+#include "usdMaya/api.h"
+#include "pxr/usd/sdf/path.h"
+#include "usdMaya/JobArgs.h"
+
+#include <maya/MDagPath.h>
+#include <maya/MObjectHandle.h>
+
+#include <memory>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+class MayaPrimWriter;
+typedef std::shared_ptr<MayaPrimWriter> MayaPrimWriterPtr;
+
+class usdWriteJobCtx {
+public:
+    PXRUSDMAYA_API
+    usdWriteJobCtx(const JobExportArgs& args);
+    const JobExportArgs& getArgs() const { return mArgs; };
+    const UsdStageRefPtr& getUsdStage() const { return mStage; };
+    // Querying the master path for instancing. This also creates the mesh if it doesn't exists.
+    SdfPath getMasterPath(const MDagPath& dg);
```

Comment:
This function should have the PXRUSDMAYA_API decoration as well since it's not implemented inline